src/linear_solver/NNLS_solver.cpp

```diff
@@ -199,9 +203,9 @@ void NNLS_solver::moveToInactiveSet(int idx){
 }
 
 bool NNLS_solver::solve(){
+  int rank = this->Comm_.MyPID();
```

Comment:
It's good practice to add `const` when possible.

Reply:
Changed this line and a few others in the following functions
`NNLS_solver::allocateVectorSingleCore(const Epetra_Vector &b)`
`NNLS_solver::allocateMatrixSingleCore(const Epetra_CrsMatrix &A)`
`NNLS_solver::allocateVectorSingleCore(const Epetra_Vector &c)`

Reply:
Perfect!